packages/loader/container-loader/src/container.ts

```diff
@@ -715,35 +722,36 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         }
         this._closed = true;
 
-        this.collabWindowTracker.stopSequenceNumberUpdate();
-        this._deltaManager.close(error);
-
-        this._protocolHandler?.close();
+        // Ensure that we raise all key events even if one of these throws
+        try {
+            this.collabWindowTracker.stopSequenceNumberUpdate();
+            this._deltaManager.close(error);
 
-        this._context?.dispose(error !== undefined ? new Error(error.message) : undefined);
+            this._protocolHandler?.close();
 
-        assert(this.connectionState === ConnectionState.Disconnected, 0x0cf /* "disconnect event was not raised!" */);
+            this._context?.dispose(error !== undefined ? new Error(error.message) : undefined);
 
-        this._storageService?.dispose();
+            assert(this.connectionState === ConnectionState.Disconnected,
+                0x0cf /* "disconnect event was not raised!" */);
 
-        // Notify storage about critical errors. They may be due to disconnect between client & server knowledge about
-        // file, like file being overwritten in storage, but client having stale local cache.
-        // Driver need to ensure all caches are cleared on critical errors
-        this.service?.dispose(error);
+            this._storageService?.dispose();
 
-        if (error !== undefined) {
-            this.logger.sendErrorEvent(
-                {
-                    eventName: "ContainerClose",
-                    lastSequenceNumber: this._deltaManager.lastSequenceNumber,
-                },
-                error,
-            );
-        } else {
-            assert(this.loaded, 0x0d0 /* "Container in non-loaded state before close!" */);
-            this.logger.sendTelemetryEvent({ eventName: "ContainerClose" });
+            // Notify storage about critical errors. They may be due to disconnect between client & server knowledge
+            // about file, like file being overwritten in storage, but client having stale local cache.
+            // Driver need to ensure all caches are cleared on critical errors
+            this.service?.dispose(error);
+        } catch (exception) {
+            this.logger.sendErrorEvent({ eventName: "ContainerCloseException"}, exception);
         }
 
+        this.logger.sendErrorEvent(
```

Comment:
```suggestion
        this.logger.sendTelemetryEvent(
```

This will set the category to Error if `error` is not undefined

Reply:
Good point, will change,